src/main/java/dev/jbang/Script.java

```diff
@@ -274,6 +279,15 @@ public String resolveClassPath(boolean offline) {
 				}
 			} else {
 				List<String> dependencies = collectDependencies();
+				if (getResolvedSourcePaths() != null) {
+					for (Path resource : getResolvedSourcePaths()) {
+						try {
+							dependencies.addAll(collectDependencies(Files.readAllLines(resource)));
```

Comment:
this would be loading all strings into memory - we shouldn't need that.....maybe not big issue now but something to defintely look at down the road to avoid too much GC

Reply:
I change to use Files.lines instead.
Do you think is it better now? Or do you recommend another approach?

```diff
@@ -120,28 +120,27 @@ public class Script {
        }
 
        public List<String> collectDependencies() {
-               return collectDependencies(getLines());
+               return collectDependencies(getLines().stream());
        }
 
-       public List<String> collectDependencies(List<String> lines) {
+       public List<String> collectDependencies(Stream<String> lines) {
                if (forJar()) { // if a .jar then we don't try parse it for dependencies.
                        return additionalDeps;
                }
                // early/eager init to property resolution will work.
                new Detector().detect(new Properties(), Collections.emptyList());
 
-               // Make sure that dependencies declarations are well formatted
-               if (lines.stream().anyMatch(it -> it.startsWith("// DEPS"))) {
-                       throw new IllegalArgumentException("Dependencies must be declared by using the line prefix //DEPS");
-               }
-
                Properties p = new Properties(System.getProperties());
                if (properties != null) {
                        p.putAll(properties);
                }
 
-               Stream<String> depStream = lines.stream()
-                                                                               .filter(it -> isDependDeclare(it))
+               Stream<String> depStream = lines.filter(it -> {
+                       // Make sure that dependencies declarations are well formatted
+                       if (it.startsWith("// DEPS"))
+                               throw new IllegalArgumentException("Dependencies must be declared by using the line prefix //DEPS");
+                       return isDependDeclare(it);
+               })
                                                                                .flatMap(it -> extractDependencies(it))
                                                                                .map(it -> PropertiesValueResolver.replaceProperties(it, p));
 
@@ -282,7 +281,7 @@ public class Script {
                                if (getResolvedSourcePaths() != null) {
                                        for (Path resource : getResolvedSourcePaths()) {
                                                try {
-                                                       dependencies.addAll(collectDependencies(Files.readAllLines(resource)));
+                                                       dependencies.addAll(collectDependencies(Files.lines(resource)));
                                                } catch (IOException e) {
                                                        throw new RuntimeException(e);
                                                }
```